src/new/templates/rust/ui/hyperapp/skeleton-app/src/lib.rs

```diff
@@ -0,0 +1,163 @@
+// HYPERWARE SKELETON APP
+// This is a minimal, well-commented skeleton app for the Hyperware platform
+// using the Hyperapp framework (macro-driven approach).
+
+// CRITICAL IMPORTS - DO NOT MODIFY THESE
+// The hyperprocess_macro provides everything you need including:
+// - Async/await support (custom runtime)
+// - Automatic WIT (WebAssembly Interface Types) generation
+// - State persistence
+// - HTTP/WebSocket bindings
+use hyperprocess_macro::hyperprocess;
+
+// HYPERWARE PROCESS LIB IMPORTS
+// These are provided by the hyperprocess_macro, DO NOT add hyperware_process_lib to Cargo.toml
+use hyperware_process_lib::{homepage::add_to_homepage, our, println};
+
+// Standard imports for serialization
+use serde::{Deserialize, Serialize};
+
+// STEP 1: DEFINE YOUR APP STATE
+// This struct holds all persistent data for your app
+// It MUST derive Default, Serialize, and Deserialize
+// Add PartialEq if you use this type in WIT interfaces
+#[derive(Default, Serialize, Deserialize)]
+pub struct AppState {
+    // Example fields - replace with your app's data
+    counter: u32,
+    messages: Vec<String>,
+}
+
+// STEP 2: IMPLEMENT YOUR APP LOGIC
+// The #[hyperprocess] attribute goes HERE, before the impl block
+#[hyperprocess(
+    name = "Skeleton App",
+    ui = Some(hyperware_process_lib::http::server::HttpBindingConfig::default()),
+    endpoints = vec![
+        hyperware_process_lib::hyperapp::Binding::Http {
+            path: "/api",
+            config: hyperware_process_lib::http::server::HttpBindingConfig::new(false, false, false, None),
+        },
+    ],
+    save_config = hyperware_process_lib::hyperapp::SaveOptions::EveryMessage,
+    wit_world = "skeleton-app-dot-os-v0"
+)]
+impl AppState {
+    // INITIALIZATION FUNCTION
+    // Runs once when your process starts
+    // Use this to:
+    // - Register with the homepage
+    // - Set up initial state
+    // - Connect to other system processes
+    #[init]
+    async fn initialize(&mut self) {
+        // Add your app to the Hyperware homepage
+        // Parameters: name, icon (emoji), path, widget
+        add_to_homepage("Skeleton App", Some("🦴"), Some("/"), None);
+        
+        // Initialize your app state
+        self.counter = 0;
+        self.messages.push("App initialized!".to_string());
+        
+        // Get our node identity (useful for P2P apps)
+        let our_node = our().node.clone();
+        println!("Skeleton app initialized on node: {}", our_node);
+    }
+    
+    // HTTP ENDPOINT EXAMPLE
+    // CRITICAL: ALL HTTP endpoints MUST accept _request_body parameter
+    // even if they don't use it. This is a framework requirement.
+    #[http]
+    async fn get_status(&self, _request_body: String) -> String {
```

Comment:
remove arg. Change the return value to a rust type. change the function to return that rust type (do not serialize it -- the framework does it under the hood). Update comment accordingly